src/test/java/com/palantir/gradle/shadowjar/ShadowJarPluginIntegrationSpec.java

```diff
@@ -0,0 +1,626 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.ctc.wstx.stax.WstxInputFactory;
+import com.ctc.wstx.stax.WstxOutputFactory;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.google.common.io.CharStreams;
+import com.palantir.gradle.testing.execution.GradleInvoker;
+import com.palantir.gradle.testing.execution.InvocationResult;
+import com.palantir.gradle.testing.junit.GradlePluginTests;
+import com.palantir.gradle.testing.maven.MavenArtifact;
+import com.palantir.gradle.testing.maven.MavenRepo;
+import com.palantir.gradle.testing.project.RootProject;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+@GradlePluginTests
+class ShadowJarPluginIntegrationSpec {
+    private static final XmlMapper MAPPER = new XmlMapper(new WstxInputFactory(), new WstxOutputFactory());
+    private static final String MAVEN_ROOT = "build/repo";
+
+    @BeforeEach
+    void setup(RootProject rootProject) {
+        rootProject.settingsGradle().rootProjectName("asd-fgh");
+
+        rootProject
+                .buildGradle()
+                .plugins()
+                .add("com.palantir.consistent-versions")
+                .add("nebula.maven-nebula-publish")
+                .add("com.palantir.shadow-jar");
+        rootProject.buildGradle().append("""
+            group = 'com.palantir.bar-baz_quux'
+            version = '2'
+
+            repositories {
+                mavenCentral()
+            }
+
+            publishing {
+                repositories {
+                    maven {
+                        name 'testRepo'
+                        url "$projectDir/%s"
+                    }
+                }
+            }
+            """, MAVEN_ROOT);
+    }
+
+    @Test
+    void when_using_shadeTransitively_produced_pom_has_only_non_shaded_dependencies(
+            GradleInvoker gradle, RootProject project) throws IOException {
+        project.buildGradle().append("""
+            dependencies {
+                api 'org.checkerframework:checker-qual:2.10.0'
+
+                shadeTransitively 'com.google.guava:guava:28.2-jre'
+            }
+            """);
+        project.buildGradle().plugins().add("java-library");
+
+        gradle.withArgs("publishNebulaPublicationToTestRepoRepository", "--warning-mode=none", "--write-locks")
```

Comment:
what's the reason for adding `--warning-mode=none` ? we didn't have this previously